src/program.cpp

```diff
@@ -52,6 +52,32 @@ struct membuf : public std::streambuf {
         auto send = reinterpret_cast<char*>(end);
         setp(sbegin, send);
     }
+    virtual pos_type seekoff(off_type off, std::ios_base::seekdir dir,
+                             std::ios_base::openmode which) {
+        if (off != 0)
+            return -1;
+        char* whence = eback();
+        if (dir == std::ios_base::cur) {
+            whence = gptr();
+        } else if (dir == std::ios_base::end) {
+            whence = egptr();
+        }
+        char* to = whence + off;
+        if (to >= eback() && to <= egptr()) {
```

Comment:
Should this be `to < egptr()`? Seems `egptr` returns a pointer _past_ the end of the get area (https://en.cppreference.com/w/cpp/io/basic_streambuf/gptr).

Reply:
Do we not want to allow to seek to the end of the stream?
For example, do you think seekoff(0, ios_base::end...) should return -1?

Reply:
Yes, we do and what you've done is correct. I was off by one on the first review :).